homeassistant/components/ezviz/number.py

```diff
@@ -53,46 +86,52 @@ def __init__(
         self,
         coordinator: EzvizDataUpdateCoordinator,
         serial: str,
-        sensor: str,
-        description: NumberEntityDescription,
+        description: EzvizNumberEntityDescription,
+        value: str,
     ) -> None:
         """Initialize the sensor."""
         super().__init__(coordinator, serial)
-        self._sensor_name = sensor
-        self.battery_cam_type = bool(
-            self.data["device_category"]
-            == DeviceCatagories.BATTERY_CAMERA_DEVICE_CATEGORY.value
-        )
-        self._attr_unique_id = f"{serial}_{sensor}"
-        self._attr_native_max_value = 100 if self.battery_cam_type else 6
+        self.sensitivity_type = 3 if value == "3" else 0
+        self._attr_native_max_value = 100 if value == "3" else 6
+        self._attr_unique_id = f"{serial}_{description.key}"
         self.entity_description = description
+        self.sensor_value: int | None = None
 
     @property
     def native_value(self) -> float | None:
         """Return the state of the entity."""
-        try:
-            return float(self.data[self._sensor_name])
-        except ValueError:
-            return None
+        if self.sensor_value is not None:
+            return float(self.sensor_value)
+        return None
 
     def set_native_value(self, value: float) -> None:
         """Set camera detection sensitivity."""
         level = int(value)
         try:
-            if self.battery_cam_type:
-                self.coordinator.ezviz_client.detection_sensibility(
-                    self._serial,
-                    level,
-                    3,
-                )
-            else:
-                self.coordinator.ezviz_client.detection_sensibility(
-                    self._serial,
-                    level,
-                    0,
-                )
+            self.coordinator.ezviz_client.detection_sensibility(
+                self._serial,
+                level,
+                self.sensitivity_type,
+            )
 
         except (HTTPError, PyEzvizError) as err:
             raise HomeAssistantError(
                 f"Cannot set detection sensitivity level on {self.name}"
             ) from err
+
+        self.sensor_value = level
+
+    def update(self) -> None:
+        """Fetch data from EZVIZ."""
+        _LOGGER.debug("Updating %s", self.name)
+        try:
+            self.sensor_value = self.coordinator.ezviz_client.get_detection_sensibility(
+                self._serial,
+                str(self.sensitivity_type),
+            )
+
+        except (EzvizAuthTokenExpired, EzvizAuthVerificationCode) as error:
+            raise ConfigEntryAuthFailed from error
```

Comment:
Nothing will handle `ConfigEntryAuthFailed` raised here. I suggest reloading the config entry instead to stop updating in vain until the auth error is resolved and take advantage of the default back-off logic in place. It will also correctly make the entities unavailable.

Example:
https://github.com/home-assistant/core/blob/be2389a9dbaa7be276a7311dd1c8ff63ef1a3a30/homeassistant/components/vlc_telnet/media_player.py#L127-L132

Reply:
Added logic to reload the entity as suggested.